### PR TITLE
[SQL] Add trigonometric functions sin() and cos()

### PR DESCRIPTION
**Pull request by abhizer**
* Also changes the precision to which floats display

Fixes: #1094

Is this a user-visible change (yes/no): yes

### CODE REVIEW

File: docs/sql/float.md [hunk 1]
```diff
@@ -84,4 +84,12 @@ REAL '1.23'  -- string style
     <td><code>LOG10(value)</code></td>
     <td>The logarithm base 10 of value. Produces a runtime error for values less than or equal to zero.</td>
   </tr>
+  <tr>
+    <td><code>SIN(value)</code></td>
```
Comment (mihaibudiu) [inline docs/sql/float.md:88]: You have to document the fact that the only type supported is double, and everything else is cast to double, and also that the result is always double. We will have to revisit the other functions, like ln too. Maybe you can file an issue about that.

File: crates/dbsp/src/algebra/floats.rs [hunk 2]
```diff
@@ -361,7 +364,9 @@ macro_rules! float {
             impl Display for $outer {
                 #[rustfmt::skip]
                 fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                    Display::fmt(&self.0.0, f)
+                    let num = format!("{1:.0$}", FLOAT_DISPLAY_PRECISION, &self.0.0);
```
Comment (mihaibudiu) [inline crates/dbsp/src/algebra/floats.rs:367]: Let's do this in the cast to string method, which is in `sqllib/src/casts.rs`

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java [hunk 3]
```diff
@@ -624,6 +624,13 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                         return this.compilePolymorphicFunction(call, node, type,
                                 ops, 2);
                     }
+                    case "sin":
+                    case "cos":
+                    {
+                        // Turn the argument into Double
+                        ops.set(0, ops.get(0).cast(new DBSPTypeDouble(node, false)));
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java:631]: The `false` here is not correct, it indicates that the value cannot be null.
You should instead get the current type of the operand and copy its mayBeNull value.
This also proves that there is no test for NULL as an argument, so add that too.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
+package org.dbsp.sqlCompiler.compiler.sql.simple;
+
+import org.dbsp.sqlCompiler.compiler.sql.SqlIoTest;
+import org.junit.Test;
+
+public class TrigonometryTests extends SqlIoTest {
+    @Test
+    public void testSin() {
+        this.qs(
+                "SELECT sin(0);\n" +
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java:10]: All these values are decimal values.
You probably should write the same test with double values as arguments.
And replace 3.14 with pi when we have implemented that.
Maybe you can push another commit in this PR for PI.

File: sql-to-dbsp-compiler/lib/sqllib/src/lib.rs [hunk 5]
```diff
@@ -889,6 +889,24 @@ some_polymorphic_function1!(sign, f, F32, F32);
 some_polymorphic_function1!(sign, d, F64, F64);
 some_polymorphic_function1!(sign, decimal, Decimal, Decimal);
 
+///////////////////// sin //////////////////////
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/lib/sqllib/src/lib.rs:892]: this many comments are overkill.
Just add one comment for all trigonometric functions

**Review by mihaibudiu — CHANGES_REQUESTED** [review]
We should use 15 digits of precision for doubles and 6 for floats when converting to strings.
The formatter will also do some rounding itself, I hope the results will come out right.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java (hunk 4 above)
Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java:10]: you should also document where the tests are coming from. For example "tested using Postgres".

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+                " 0.0015926529164868282\n" +
+                "(1 row)"
+        );
+    }
+
+    // Tested using Postgres 15.2
+    @Test
+    public void testSinDouble() {
+        this.qs(
+                "SELECT sin(CAST(0 AS DOUBLE));\n" +
```
Comment (mihaibudiu) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TrigonometryTests.java:59]: this works, but you can get a double by writing `e0` at the end. `3.14e0` is a double, while `3.14` is a decimal.

**Review by mihaibudiu — APPROVED** [review]
